src/store.cc

```diff
@@ -302,10 +307,18 @@ StoreEntry::storeClientType() const
     if (swap_status == SWAPOUT_NONE)
         return STORE_MEM_CLIENT;
 
+    // TODO: The above "must make this a mem client" logic contradicts "Slight
+    // weirdness" logic in store_client::doCopy() that converts hits to misses
+    // on startSwapin() failures. We should probably attempt to open a swapin
+    // file _here_ instead (and avoid STORE_DISK_CLIENT designation for clients
+    // that fail to do so). That would also address a similar problem with Rock
+    // store that does not yet support swapin during SWAPOUT_WRITING.
```

Comment:
Please evaluate this TODO: Do you agree that there is a problem here? Do you agree with the proposed resolution, which can be summarized as "if store_client might need to open the swapin file, it must attempt to open it _before_ returning any storeClientCopy() results"? These questions have a low priority.

Reply:
Agreed with this TODO. Note however, that if we call startSwapin() here (assuming disk configuration is on), it will fail because the entry is being written on disk (it is STORE_PENDING) and Ipc::StoreMap::peekAtReader() returns nil in this case. Anyway, calling it here and marking such entry with STORE_MEM_CLIENT (instead of STORE_DISK_CLIENT and calling it later in store_client::doCopy()) looks a better idea. Probably this change will fix the TCP_SWAPFAIL_MISS logging problem described in a Jira issue comment.

Reply:
> Note however, that if we call startSwapin() here (assuming disk configuration is on), it will fail because the entry is being written on disk (it is STORE_PENDING) and Ipc::StoreMap::peekAtReader() returns nil in this case.

Yes, we will fail to start swapin for the current rock implementation, and that is better than failing later. We may succeed for UFS.